config/initializers/devise.rb

```diff
@@ -251,7 +251,8 @@
   # Add a new OmniAuth provider. Check the wiki for more information on setting
   # up on your models and hooks.
   # config.omniauth :github, 'APP_ID', 'APP_SECRET', scope: 'user,public_repo'
-
+  config.omniauth :facebook, ENV['FACEBOOK_APP_ID'], ENV['FACEBOOK_APP_SECRET'],
+                  callback_url: "http://localhost:3000/users/auth/facebook/callback"
```

Comment:
2 out of 3 -- you're going to want to be able to change this url for production, better to pull it out to the ENV as well